tests/stdlib/tsystem.nim

```diff
@@ -74,3 +74,30 @@ template main =
 
 static: main()
 main()
+
+# bug #19967
+block:
+  type
+    X = object
+      a: string
+      b: set[char]
+
+  var y = X(b: {'a'})
+
+  reset(y)
+
+  doAssert y.b == {}
+
+block:
+  type
+    Color = enum
+      Red, Blue, Yellow
+    X = object
+      a: string
+      b: set[Color]
+
+  var y = X(b: {Red, Blue})
+
+  reset(y)
+  doAssert y.b == {}
+
```

Comment:
Would be better to iterate all primitives from system, call `reset`, and assert results.

A `reset`'d type should be equal to `default()` of that type for easy assert (?) AFAIK.

Reply:
I agree it can be a following-up pr.